modules/solid_mechanics/src/materials/NewtonIterationSingleVariable.C

```diff
@@ -0,0 +1,443 @@
+/****************************************************************/
+/* MOOSE - Multiphysics Object Oriented Simulation Environment  */
+/*                                                              */
+/*          All contents are licensed under LGPL V2.1           */
+/*             See LICENSE for full restrictions                */
+/****************************************************************/
+#include "NewtonIterationSingleVariable.h"
+
+#include "InputParameters.h"
+#include <cmath>
+#include "Conversion.h"
+
+template <>
+InputParameters
+validParams<NewtonIterationSingleVariable>()
+{
+  InputParameters params = emptyInputParameters();
+
+  // Newton iteration control parameters
+  params.addParam<unsigned int>("max_its", 200, "Maximum number of Newton iterations");
+  params.addParam<bool>(
+      "output_iteration_info", false, "Set true to output Newton iteration information");
+  params.addDeprecatedParam<bool>(
+      "output_iteration_info_on_error",
+      false,
+      "Set true to output Newton iteration information when those iterations fail",
+      "This information is always output when iterations fail");
+  params.addParam<Real>(
+      "relative_tolerance", 1e-8, "Relative convergence tolerance for Newton iteration");
+  params.addParam<Real>(
+      "absolute_tolerance", 1e-11, "Absolute convergence tolerance for Newton iteration");
```

Comment:
I agree with your point, but this is a general base class that isn't intended to prescribe the specifics of that.  That's something determined by the models that derive from this.